e2e/common/cli/delete_test.go

```diff
@@ -0,0 +1,79 @@
+
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	. "github.com/apache/camel-k/e2e/support"
+)
+
+func TestKamelCLIDelete(t *testing.T) {
+	WithNewTestNamespace(t, func(ns string) {
+		Expect(Kamel("install", "-n", ns).Execute()).To(Succeed())
+
+		t.Run("delete running integration", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "../files/yaml.yaml").Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, "yaml"), TestTimeoutMedium).Should(Equal(v1.PodRunning))
+			Expect(Kamel("delete", "yaml", "-n", ns).Execute()).To(Succeed())
+			Expect(GetOutputString(Kamel("get", "-n", ns))).ShouldNot(ContainSubstring("yaml"))
+		})
+
+		t.Run("delete building integration", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "../files/yaml.yaml").Execute()).To(Succeed())
+			Expect(Kamel("delete", "yaml", "-n", ns).Execute()).To(Succeed())
+			Expect(GetOutputString(Kamel("get", "-n", ns))).ShouldNot(ContainSubstring("yaml"))
+		})
+
+		t.Run("delete integration from csv", func(t *testing.T) {
+			Expect(Kamel("run", "https://github.com/apache/camel-k/blob/master/e2e/common/files/yaml.yaml", "-n", ns).Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, "yaml"), TestTimeoutMedium).Should(Equal(v1.PodRunning))
+			Expect(Kamel("delete", "yaml", "-n", ns).Execute()).To(Succeed())
+			Expect(GetOutputString(Kamel("get", "-n", ns))).ShouldNot(ContainSubstring("yaml"))
+		})
+
+		t.Run("delete several integrations", func(t *testing.T) {
+			Expect(Kamel("run", "../files/yaml.yaml", "-n", ns).Execute()).To(Succeed())
+			Expect(Kamel("run", "../files/Java.java", "-n", ns).Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, "yaml"), TestTimeoutMedium).Should(Equal(v1.PodRunning))
+			Eventually(IntegrationPodPhase(ns, "java"), TestTimeoutMedium).Should(Equal(v1.PodRunning))
+			Expect(Kamel("delete", "yaml", "-n", ns).Execute()).To(Succeed())
+			Expect(GetOutputString(Kamel("get", "-n", ns))).ShouldNot(ContainSubstring("yaml"))
+			Expect(Kamel("delete", "java", "-n", ns).Execute()).To(Succeed())
+			Expect(GetOutputString(Kamel("get", "-n", ns))).ShouldNot(ContainSubstring("java"))
```

Comment:
@vsokolov00 Maybe you should rather verify that the integration resource and the pod were actually removed.